bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/BufferedChannel.java

```diff
@@ -88,8 +90,12 @@ public BufferedChannel(FileChannel fc, int writeCapacity, int readCapacity, long
 
     @Override
     public synchronized void close() throws IOException {
+        if (closed) {
```

Comment:
why is it being called more than one time in the first place?
Can we address that issue (root cause) instead of providing work around?

Reply:
it can probably double closed because the way we managed `try() finally {}` at shutting down entry log manager. we can fix the logic there. however I don't think that is a redirection to go, because if close() is being called twice in different places, the problem still exists. so I went down to this path to make close idempotent in the base class, which it will address the problem of now and any potentially problems in future.

Reply:
I agree with Sikie

Reply:
ok, accepted.